backend/src/test/java/com/ambacam/rest/typerequetes/TypeRequetesResourceIT.java

```diff
@@ -0,0 +1,159 @@
+package com.ambacam.rest.typerequetes;
+
+import com.ambacam.ItBase;
+import com.ambacam.model.TypeRequete;
+import com.ambacam.repository.TypeRequeteRepository;
+import com.ambacam.rest.ApiConstants;
+import io.restassured.http.ContentType;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import static io.restassured.RestAssured.given;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.*;
+
+public class TypeRequetesResourceIT extends ItBase {
+
+	@Autowired
+	private TypeRequeteRepository repository;
+
+	private TypeRequete typeRequete1;
+
+	private TypeRequete typeRequete2;
+
+	@Before
+	public void setup() throws Exception {
+		super.setup();
+
+		// create typeRequete1
+		typeRequete1 = buildTypeRequete();
+		typeRequete1 = repository.save(typeRequete1);
+		// create typeRequete2
+		typeRequete2 = repository.save(buildTypeRequete());
+	}
+
+	@Override
+	@After
+	public void cleanup() throws Exception {
+		repository.deleteAll();
+		super.cleanup();
+	}
+
+	@Test
+	public void create() {
+		TypeRequete create = buildTypeRequete();
+		int id = given().contentType(ContentType.JSON).body(create).log().body()
+				.post(ApiConstants.TYPE_REQUETE_COLLECTION).then().log().body().statusCode(200).extract().body()
+				.path("id");
+
+		// check that the typeRequete has been saved
+		TypeRequete actual = repository.findOne(Integer.toUnsignedLong(id));
+		assertThat(actual, is(notNullValue()));
+		assertThat(actual.getNom(), is(equalTo(create.getNom())));
+		assertThat(actual.getDescription(), is(equalTo(create.getDescription())));
+	}
+
+	@Test
+	public void createNomNull() {
+		TypeRequete create = buildTypeRequete();
+		create.setNom(null);
+		given().contentType(ContentType.JSON).body(create).log().body().post(ApiConstants.TYPE_REQUETE_COLLECTION)
+				.then().log().body().statusCode(400);
+	}
+
+	@Test
+	public void createNomVide() {
+		TypeRequete create = buildTypeRequete();
+		create.setNom("");
+		given().contentType(ContentType.JSON).body(create).log().body().post(ApiConstants.TYPE_REQUETE_COLLECTION)
+				.then().log().body().statusCode(400);
+	}
+
+	@Test
+	public void createMemeNom() {
+		TypeRequete create = buildTypeRequete();
+		create.setNom(typeRequete1.getNom());
+		given().contentType(ContentType.JSON).body(create).log().body().post(ApiConstants.TYPE_REQUETE_COLLECTION)
+				.then().log().body().statusCode(400);
+	}
+
+	@Test
+	public void list() {
+		given().get(ApiConstants.TYPE_REQUETE_COLLECTION).then().log().body().statusCode(200)
+				.body("size()", is(equalTo(2)))
+				.body("id", containsInAnyOrder(typeRequete1.getId().intValue(), typeRequete2.getId().intValue()));
```

Comment:
Please check all parameters of one typeRequete

Reply:
thx